utils/pretty_printers.cc

```diff
@@ -11,16 +11,21 @@
 namespace utils {
 
 std::ostream& operator<<(std::ostream& os, pretty_printed_data_size data) {
-    static constexpr const char *suffixes[] = {" bytes", "kB", "MB", "GB", "TB", "PB"};
-
-    unsigned exp = 0;
-    while ((data._size >= 1000) && (exp < sizeof(suffixes))) {
-        exp++;
-        data._size /= 1000;
+    static constexpr auto suffixes = {" bytes", "kB", "MB", "GB", "TB", "PB"};
```

Comment:
Nitpick: This "auto" is very mysterious, I liked better the explicit type. But it's a matter of style, so I'm ok with this change.

Reply:
P.S. the code below converts these C constants into string_view, which isn't free (requires strlen()). Perhaps you can work below with "const char*", or alternatively, prepare "suffixes" as an array of std::string_view instead of character pointers?

Reply:
> prepare "suffixes" as an array of std::string_view instead of character pointers?

another Nadav would argue it's not free either. `string_view` keeps two things: a `const char*` pointer and a `size_t` length.

Reply:
i will go with a `const char*`.

Reply:
> > prepare "suffixes" as an array of std::string_view instead of character pointers?
> 
> another Nadav would argue it's not free either. `string_view` keeps two things: a `const char*` pointer and a `size_t` length.

Yes, this was why it was my second, "alternatively", suggestion :-)